### PR TITLE
fix: add `vrfDelay` as a `phase` return value for ongoing auctions to reflect VRFdelays in polkadot

### PR DESCRIPTION
**Pull request by TarikGul**
closes: [#588](https://github.com/paritytech/substrate-api-sidecar/issues/588)

### Summary

This PR aims to reflect the addition of the `AuctionStatus` enum in polkadot for parachains, specifically the `VRFDelay`. When an auction is past its `finishEnd` block the auction is then in a `vrfDelay` phase. 

### Changes

* I updated the tests to reflect this change, I also spotted a minor issue in the tests and reversed the output results to reflect a real world scenario with the auctions.

* Added logic inside of the `auctionsCurrent` method for `ParasService` that will add a `vrfDelay` phase as a potential result.

* I added the `vrfDelay` result to the `AuctionPhase` type, and changed `starting`=> `startPeriod`, `ending` => `endPeriod`. 

Overall there are 3 potential return values for a phase:

```
phase :
  startPeriod
  endPeriod
  vrfDelay
```

* Updated the docs

### CODE REVIEW

File: src/services/paras/ParasService.ts
Comment (emostov) [inline src/services/paras/ParasService.ts:243]:
```suggestion
				phase = 'vrfDelay';
```

File: docs/src/openapi-v1.yaml
Comment (emostov) [inline docs/src/openapi-v1.yaml:2349]:
```suggestion
            An auction can be in one of 4 phases. Both `startingPeriod` () and `endingPeriod` indicate
            an ongoing auction, while `vrfDelay` lines up with the `AuctionStatus::VrfDelay` . Finally, a value of `null`
            indicates there is no ongoing auction. Keep in mind the that the `finishEnd` field is the block number the 
            `endingPeriod` finishes and the `vrfDelay` period begins. The `vrfDelay` period is typically about an 
            epoch long and no crowdloan contributions are accepted. 
```

File: docs/src/openapi-v1.yaml
Comment (emostov) [inline docs/src/openapi-v1.yaml:2343]:
```suggestion
          - startingPeriod
          - endingPeriod
          - vrfDelay
```
I think we should maximize alignment with https://github.com/paritytech/polkadot/blob/60341e125563cbf655ab686ac5cef370ac33d16d/runtime/common/src/traits.rs#L150-L164

File: src/services/paras/ParasService.spec.ts
Comment (emostov) [inline src/services/paras/ParasService.spec.ts:191]:
```suggestion
				phase: 'vrfDelay',
```

File: src/services/paras/ParasService.spec.ts
Comment (emostov) [inline src/services/paras/ParasService.spec.ts:241]:
```suggestion
			const expectedResponse = 'endingPeriod';
```

File: src/services/paras/ParasService.ts
Comment (emostov) [inline src/services/paras/ParasService.ts:245]:
```suggestion
				phase = beginEnd.gt(blockNumber) ? 'startingPeriod' : 'endingPeriod';
```

File: src/types/responses/Paras.ts
Comment (emostov) [inline src/types/responses/Paras.ts:15]:
```suggestion
export type AuctionPhase = 'startingPeriod' | 'endingPeriod' | 'vrfDelay';
```

**Review by emostov — APPROVED** [review]
lgtm - just some suggestions on docs and the name of the enum variants .. although i am not sure if appending `Period` is worth it to align with better with https://github.com/paritytech/polkadot/blob/60341e125563cbf655ab686ac5cef370ac33d16d/runtime/common/src/traits.rs#L150-L164

File: docs/src/openapi-v1.yaml
Comment (dvdplm) [reply docs/src/openapi-v1.yaml:2343]: > i will not that im not sure what to do for casing on `vrfDelay`

I think `vrfDelay` is fine. It could be `VRFDelay` as well, given it's an acronym? But `vrfDelay` sounds better to me.

Comment (dvdplm) [reply docs/src/openapi-v1.yaml:2343]: I agree we should stay close to the Rust source. I also think it's a bit unfortunate we went with `-ing` rather than `StartPeriod`/`EndPeriod`.

**Review by dvdplm — APPROVED** [review]

File: docs/src/openapi-v1.yaml
Comment (TarikGul) [reply docs/src/openapi-v1.yaml]: Yea I like the idea of `vrfDelay`. One thing I just want to confirm too, if we append `Period` to `starting` and `ending` it will be a breaking change for the API. Just something to note. If you both think its appropriate while its still an experimental endpoint I am down to make the change.

**Comment by TarikGul** [comment]
@emostov This is ready to merge, just want to make sure whether we should append `Period` and change this PR to a breaking change, or keep as if and just add the `vrfDelay`

**Comment by emostov** [comment]
> @emostov This is ready to merge, just want to make sure whether we should append `Period` and change this PR to a breaking change, or keep as if and just add the `vrfDelay`

Its already a breaking change because when we used to show `ending` we will now sometimes show `vrfDelay`. IMO we should change `ending` => `endPeriod` & `starting` => `startPeriod` to better align with the `AuctionStatus` enum

**Comment by TarikGul** [comment]
Ahh yes, the breaking change is there regardless because the new runtime logic is keeping `AuctionInfo` alive during the `vrfDelay` period. Good point. Updated `starting` => `startPeriod`, `ending` => `endPeriod` and reflected it in the docs. I will set up a release after I merge this in once the tests pass.

**Pull request merged**
